MzmlReader/MzmlReader.cs

```diff
@@ -50,7 +50,7 @@ public Run LoadMzml(string path, double massTolerance, bool storeScansInMemory,
                 }
                 catch (IOException ex)
                 {
-                    string exceptionInfo = ("Unable to open the file {path}. Please close the application using it and try again.");
+                    string exceptionInfo = ("Unable to open the file "+ path + ". Is it perhaps in use?");

```

Comment:
So a few things about how we should handle these. 

- We still want to preserve the stack trace but maybe we can also give the logger info to the user in a friendly format. Creating a new exception here is an antipattern as we aren't preserving the stack trace (bad).
- Always prefer string interpolation with String.Format to concatenating strings like this.

logger.Error(String.Format("Unable to open the file{0}. Is it perhaps in use?", path);
throw ex;

Reply:
Can I drop a few points in to do with NLog specifically, and exception handling in libraries more generally?

* NLog provides functions for logging exceptions; please always use them, as they give the ability to log the stack trace, message, and many other things.  So logger.Error(ex, message).
* NLog has built-in string.Format functionality; just pass a format string and more parameters.  This is deliberate, as NLog is designed to be high-performance and it will not do the format if the error will not be logged.  So logger.Error (ex, "I want to say something about {0}, please", path) is preferred.

More generally:

* MzmlReader is designed to be called from other code: it's a library.  Where you're not *absolutely certain* you know what an exception means and how you should handle it, the general rule is simple: *don't handle it*.  In libraries, you almost certainly don't know how the calling application wants you to handle the exception, so don't even catch it.  Let it bubble up the stack to something that knows what it's doing, or - if you really, really want to, and I don't recommend it - create a custom exception type that contains more context *and the inner exception* and throw that from your catch point.  I don't recommend that because it's painful for me as a library user.  I can't just break on exceptions and see where in your code there's a problem; instead I have to unwrap an exception, eyeball the inner exception, and guess.  Deeply unpleasant.  Windows Communication Fondation threw layers and layers of wrapped exceptions, and was one of the worst libraries it's ever been my misfortune to debug.
* If you want to log an exception and then re-throw it, *always* use "throw" not "throw ex".  The former keeps the stack trace intact; the latter starts a new stack trace from the current location and therefore loses vital context.

Reply:
As ever I defer to Peter's comments - I made a few rookie errors in my PR feedback.

In general, the take-home from all of this is this: letting things fail and preserving the stack trace is always preferable to swallowing an error (deeply unhelpful to another developer using your code) and allowing unintended code execution afterwards.

Reply:
That makes total sense!

I've realized that we can check whether we can try to open the file in
Yamato.Console already so the mzmlparser library doesn't have to.
I've moved it to the console now. Also - I am opening the file just to
check if its possible so we have a specific error message. Then we actually
only load it in the next function. Is that fine? It seemed the easiest way
to figure out exactly whether or not the problem was the file being able to
open or not.